### PR TITLE
Account Settings Page

### LINKED ISSUE #141: Account Settings Page
- View/Edit E-Mail used
- View Plan (Free,Hobby,Startup,etc)
- Change Plan

### PR DESCRIPTION
**Pull request by MrF3lix**
## Related Issues

- Closes #141 

## Definition of Done

- [x] User Story
  - [x] All Acceptance Criteria met
- [x] Code
  - [x] No more code needed
  - [x] No known bugs
- [x] Clean Code
  - [x] Comments have been created for new code
  - [x] ESLinter found no issues (editor & executor only)
  - [x] No unavoidable Code Smells
- [x] Testing
  - [x] Existing Tests pass
  - [x] New Unit Tests created
  - [x] New Unit Tests pass
- [x] SonarCloud Quality Gate passed

### CODE REVIEW

File: src/editor/pages/account.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                  <Input
+                    label="Username"
+                    onChange={( e ): void => setUserName( e.target.value )}
+                    value={userName}
+                  />
+                </div>
+              </FormGroup>
+            </FormList>
+
+            <FormList title="Plan & Billing" description="This information describes your plan and billing">
```
Comment (nikipuk) [inline src/editor/pages/account.tsx:112]: . missing

File: src/editor/pages/account.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+                  />
+                  <p className="mt-1 text-sm text-gray-500 dark:text-gray-400">
+                  Changing the billing option is not yet supported<br />
+                  Contact <a className="text-blue-500" href="mailto:info@flooq.io">info@flooq.io</a> for help.
+                  </p>
+                </div>
+              </FormGroup>
+            </FormList>
+
+            <FormList title="Danger Zone" description="Warning: The actions you take in this section are irreversible">
```
Comment (nikipuk) [inline src/editor/pages/account.tsx:141]: . missing

**Review by nikipuk — CHANGES_REQUESTED** [review]
I feel like it is currently harder to delete a Data Flow than your entire account. Maybe there should be another "add username/email to confirm" section.

Other than the requested minor changes, the Account Settings page looks godd and seems to be working as intended.

File: src/api/Flooq.Test/Controllers/DataFlowControllerTest.cs [hunk 3]
```diff
@@ -332,4 +332,21 @@ public async Task Delete_ReturnsNotFoundIfThereIsNoMatchingDataFlow()
     Assert.IsNotNull(actionResult);
     Assert.IsInstanceOfType(actionResult, typeof(NotFoundResult));
   }
+
+  [TestMethod]
+  public async Task CanDeleteAllDataFlow()
+  {
```
Comment (DuplosFidibuss) [inline src/api/Flooq.Test/Controllers/DataFlowControllerTest.cs:338]: Rename to CanDeleteAllDataFlows.

File: src/api/Flooq.Test/Controllers/TokenControllerTest.cs [hunk 4]
```diff
@@ -268,4 +268,19 @@ public async Task Delete_ReturnsNotFoundIfThereIsNoMatchingToken()
     Assert.IsNotNull(actionResult);
     Assert.IsInstanceOfType(actionResult, typeof(NotFoundResult));
   }
+
+  [TestMethod]
+  public async Task CanDeleteAllToken()
+  {
+    var tokenController = new TokenController(_tokenServiceMock.Object);
```
Comment (DuplosFidibuss) [inline src/api/Flooq.Test/Controllers/TokenControllerTest.cs:275]: Rename to CanDeleteAllTokens

File: src/api/Flooq.Test/Services/TokenServiceTest.cs [hunk 5]
```diff
@@ -197,4 +197,25 @@ public void TestHasUserEquallyNamedToken()
...
+    Assert.IsNotNull(token);
+    Assert.AreEqual(1, _context.Tokens.Count());
+    
+    tokenService.RemoveAllTokensByUserId(TEST_USER_ID);
+    await _context.SaveChangesAsync();
+    
+    var removedToken = await _context.Tokens.FindAsync(_token.Id);
+    Assert.IsNull(removedToken);
+    Assert.AreEqual(_token.Id, token.Id);
+    Assert.AreEqual(0, _context.Tokens.Count());
```
Comment (DuplosFidibuss) [inline src/api/Flooq.Test/Services/TokenServiceTest.cs:219]: I think the `Assert.AreEqual(_token.Id, token.Id)` belongs to the section above; I'd place it between line 210 and 211.

File: src/identity/.vscode/launch.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            "name": ".NET Core Launch (web)",
+            "type": "coreclr",
+            "request": "launch",
+            "preLaunchTask": "build",
+            "program": "${workspaceFolder}/Flooq.Identity/bin/Debug/net6.0/Flooq.Identity.dll",
+            "args": [],
+            "cwd": "${workspaceFolder}/Flooq.Identity",
+            "stopAtEntry": false,
+            "serverReadyAction": {
```
Comment (DuplosFidibuss) [inline src/identity/.vscode/launch.json:16]: Do we need the .vscode in the Repo?

File: src/identity/Flooq.Identity/Config.cs [hunk 7]
```diff
@@ -3,15 +3,26 @@
...
 
+/// <summary>
+/// Configures the Authentication Clients as well as the available Scopes and Resources
+/// </summary>
 public static class Config
 {
+  /// <summary>
+  /// Resources that can be accessed by an authenticated user.
+  /// </summary>
+  /// <value></value>
```
Comment (DuplosFidibuss) [inline src/identity/Flooq.Identity/Config.cs:14]: What's in the value tag here?

File: src/identity/Flooq.Identity/Services/UserService.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+namespace Flooq.Identity.Services;
+
+/// <summary>
+/// Responsible for handling changes to the user accounts
+/// </summary>
+public class UserService : IUserService
+{
+  private readonly UserManager<ApplicationUser> _userManager;
+  private readonly FlooqIdentityContext _context;
+
```
Comment (DuplosFidibuss) [inline src/identity/Flooq.Identity/Services/UserService.cs:15]: As we have comments in `IUserService` I think you can remove them here.

File: src/identity/Flooq.Identity/Services/UserService.cs [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  /// <summary>
+  /// Get a user by Id
+  /// </summary>
+  /// <param name="userId">Of the user to return</param>
+  /// <returns>User with the given Id</returns>
+
+  public async Task<ApplicationUser> GetUsersById(string userId)
+  {
+    return await _context.Users.Where(u => u.Id.Equals(userId)).FirstOrDefaultAsync();
+  }
```
Comment (DuplosFidibuss) [inline src/identity/Flooq.Identity/Services/UserService.cs:33]: Why FirstOrDefault? What would be the default value?

**Review by DuplosFidibuss — CHANGES_REQUESTED** [review]

File: src/identity/Flooq.Identity/Services/UserService.cs (hunk 9 above)
Comment (MrF3lix) [reply src/identity/Flooq.Identity/Services/UserService.cs:33]: The default value is null

File: src/identity/Flooq.Identity/Services/UserService.cs (hunk 8 above)
Comment (DuplosFidibuss) [reply src/identity/Flooq.Identity/Services/UserService.cs:15]: Strange ... maybe there is an issue with the sonar configuration?

**Review by nikipuk — APPROVED** [review]
I'm happy.
but you shpuld a new issue for the saver user detete

**Pull request merged**